### ISSUE TITLE
Git not working in Code, but works in CMD

### ISSUE THREAD

**Issue by tomasbonco** [Microsoft/vscode#24434]
- VSCode Version: 1.11.1
- OS Version: Windows 10

Steps to Reproduce:

When trying to pull/push/fetch/etc., I'm getting error:
```
Could not create directory '/c/Users/[username]/.ssh'.
Host key verification failed.
fatal: Could not read from remote repository.
```

It was working before update, and also works perfectly fine from cmd (standard windows command line).

**Comment by joaomoreno** [comment]
Does it really say `[username]` there, or you manually replaced your user with that string?

Which Git flavour and version are you using?

**Comment by tomasbonco** [comment]
I manually replaced, original `/c/Users/Tomáš/.ssh`.
```
git --version
git version 2.10.2.windows.1
```

Using plain git from https://git-scm.com/download/win (no flavours as far as I know).

**Comment by joaomoreno** [comment]
Does the `C:\Users\Tomáš\.ssh` folder exist?

**Comment by tomasbonco** [comment]
Not only it does exists, it contains keys and they work when I use git from command line.

**Comment by joaomoreno** [comment]
Odd that Git tries to create it. Does it work when running from the Git Bash?

**Comment by tomasbonco** [comment]
Yes, it also works from Git Bash.

**Comment by joaomoreno** [comment]
Does it work when running from CMD within Code's terminal?

Does it work if you disable preview SCM with `SCM: Disable Preview SCM`?

**Comment by tomasbonco** [comment]
Yes, it works from Code's terminal and yes, it also works when I Disable Preview SCM (thank you!).

**Comment by joaomoreno** [comment]
Are you executing as Administrator or so?

**Comment by tomasbonco** [comment]
No, as far as I know.

**Comment by joaomoreno** [comment]
Do you also have cygwin git installed?

Looks very much like #7998

**Comment by tomasbonco** [comment]
How can I check?
In Programs & Features I don't see it. Running command cygwin, leads into error (unknown command). I'm also unable to find such folder in C:/, Program Files (both of them) and git itself. I also don't remember installing it, so I think, I'm not using cygwin.

**Comment by joaomoreno** [comment]
What does `where git` in the Command Prompt return?

**Comment by tomasbonco** [comment]
To `C:\Program Files\Git\cmd\git.exe`

**Comment by joaomoreno** [comment]
I'm running out of ideas. 😕

**Comment by tomasbonco** [comment]
Plain guess: Maybe special chars in folder name when using  Preview SCM.

**Comment by joaomoreno** [comment]
Doesn't make much sense, since all other git stuff works, right? You can see diffs, etc?

**Comment by tomasbonco** [comment]
But they don't require autentification, right?

**Comment by mike-lischke** [comment]
May I add some side notes wrt SCM preview? Since the latest vscode update I get multiple `workspace.onDidOpenTextDocument` notifications in my extension. One for the real file and 2-3 others for the same filename with an additional `.git` extension appended. When I switch off SCM preview this does not happen. These additional notifications cause my extension to fail on some processing of the files and I can imagine that other extension which are not prepared for that could break and cause weird problems. So, these additional notifications should be checked.

**Comment by joaomoreno** [comment]
@mike-lischke #24033

**Comment by joaomoreno** [comment]
@tomasbonco Is this still an issue for you?

**Comment by tomasbonco** [comment]
I cannot reproduce, because I changed my environment in the meantime. At that time using Legacy implementation helped.
